pretty/src/pretty/pretty_doc.rs

```diff
@@ -34,32 +34,39 @@ pub trait PrettyDocument: Sized + Clone {
     /// result.**
     fn bounds(&self) -> Bounds;
 
-    /// Render the document onto the screen. This method behaves as if it did
+    /// Render the document onto a `PrettyPane`. This method behaves as if it did
     /// the following:
     ///
     /// 1. Pretty-print the entire document with width `width`.
-    /// 2. Transcribe the portion of the document under the screen onto the
-    /// screen. (Remember that the screen is located at `screen.region()`.)
+    /// 2. Position the document under the `PrettyPane`, aligning `doc_pos` with the `PrettyPane`'s upper left corner.
+    /// 3. Render the portion of the document under the `PrettyPane` onto the `PrettyPane`.
     ///
     /// However, this method is more efficient than that, and does an amount of
-    /// work that is (more or less) proportional to the size of the screen,
+    /// work that is (more or less) proportional to the size of the `PrettyPane`,
     /// regardless of the size of the document.
-    fn pretty_print<Screen>(&self, width: Col, screen: &mut Screen) -> Result<(), Screen::Error>
+    fn pretty_print<'a, T>(
+        &self,
+        width: Col,
+        pane: &mut PrettyPane<'a, T>,
+        doc_pos: Pos,
+    ) -> Result<(), T::Error>
     where
-        Screen: PrettyScreen,
+        T: PrettyWindow,
     {
         let root = self.root();
         let bound = Bound::infinite_scroll(width);
         let lay = Layouts::compute(&root).fit_bound(bound);
-        render(&root, screen, screen.region()?, &lay)
+        // TODO pass doc_rect to render() instead
```

Comment:
Edit this TODO:

// TODO change `doc_region: Region` to `doc_rect: Rectangle`

Reply:
done